prelude/src/stdlib/other/System.sk

```diff
@@ -287,7 +287,17 @@ fun subprocess(
   stdout: (String) -> void = _ -> void,
   stderr: (String) -> void = _ -> void,
 ): CompletedProcess {
-  p = Posix.Popen::create{args, stdout => true, stderr => true};
+  popen{args, stdout, stderr}
+}
+
+fun popen{
+  args: Array<String>,
+  env: Map<String, String> = Map[],
+  clear_env: Bool = false,
+  stdout: (String) -> void = _ -> void,
+  stderr: (String) -> void = _ -> void,
+}: CompletedProcess {
+  p = Posix.Popen::create{args, env, clear_env, stdout => true, stderr => true};
```

Comment:
Why not also `stdin` while you are at it?

Reply:
The `popen` interface should be made more general (and support for instance `cwd`, though this is notoriously cumbersome with `posix_spawn()`). For `stdin`, do you mean a boolean option to forward to `Posix.Popen::create`?

Reply:
Yes, just thinking that if it is already there in `Popen::create` then might as well support it here too.

Reply:
I agree on the principle, but I figured then we should also support "just establishing pipes" for `stdout`/`stderr` (rather than the closures being invoked upon reads). My take away is that current `popen()`/`subprocess()` APIs are incomplete and not great, and we should rethink them through.

Reply:
Ok yes, that makes sense.